### ISSUE TITLE
2nd level Maze sometimes doesent generate bedrock

### ISSUE THREAD

**Issue by VilligerKing** [TeamTwilight/twilightforest#1362]
### Forge Version

37.0.82

### Twilight Forest Version

4.0.766

### Steps to Reproduce

no idea

### What You Expected

bedrocc

### What Happened Instead

no bedrocc

### Additional Details

screene

[
![image](https://user-images.githubusercontent.com/82680956/137817117-0c57768a-78df-4859-a9fc-13cb83f0f0cd.png)
](url)

was on tf server

### Please Read and Confirm The Following

- [X] I have confirmed this bug can be replicated without the use of Optifine.
- [X] I have confirmed the details provided in this report are concise as possible and does not contained vague information (ie. Versions are properly recorded, answers to questions are clear).

**Comment by VilligerKing** [comment]
i think i might have figured out what is causing the bedrock to not generate, on both the maze in the original post and the one in this one, the boss room generated next to the wall and both dont have bedrock, in other mazes i have found where the boss room is not next to a wall, the bedrock seemed fine

![image](https://user-images.githubusercontent.com/82680956/137967818-d6aa63dd-b5f0-4439-a347-b774167a2414.png)

**Comment by VilligerKing** [comment]
i was wrong 

![image](https://user-images.githubusercontent.com/82680956/137995936-7d150ace-3ddc-4149-af54-df0b3f2cc0c1.png)

**Comment by skythegguy** [comment]
I was going to file a report on a similar issue to this one before I found it, I did a bit more investigation and found that the maze's second level doesn't generate bedrock on the South and East walls of its second floor, when it should be completely encased. The images in your screenshots have a minimap, which shows that the walls without bedrock are the south and east ones. The same is true in my own screenshots.

![image](https://user-images.githubusercontent.com/95661376/147426620-4ffa8aec-4f61-4568-90a9-5b8f9a1a956a.png)
NE Corner

![image](https://user-images.githubusercontent.com/95661376/147426632-8eb6deca-0952-4a66-b819-84b2d5eccdc4.png)
SW Corner

Additional note:
mine are from TF version 4.0.860 and on the Official server.
